var/spack/repos/builtin/packages/rocm-openmp-extras/package.py

```diff
@@ -0,0 +1,337 @@
+
+    homepage = tools_url + "/aomp"
+    url = tools_url + "/aomp/archive/rocm-uc-3.10.0.tar.gz"
+
+    maintainers = ['srekolam', 'arjun-raj-kuppala', 'estewart08']
+    version('3.10.0', sha256=versions_dict['3.10.0']['aomp'])
+    version('3.9.0', sha256=versions_dict['3.9.0']['aomp'])
+
+    depends_on('cmake@3:', type='build')
+    depends_on('mesa18~llvm@18.3:', type=('build', 'link'))
+    depends_on('py-setuptools', type='build')
+    depends_on('python@3:', type='build')
+    depends_on('perl-data-dumper', type='build')
+    depends_on('awk', type='build')
+    depends_on('elfutils', type=('build', 'link'))
+    depends_on('libffi', type=('build', 'link'))
+
+    for ver in ['3.9.0', '3.10.0']:
+        depends_on('hsakmt-roct@' + ver, type=('build', 'run'), when='@' + ver)
+        depends_on('comgr@' + ver, type='build', when='@' + ver)
+        depends_on('hsa-rocr-dev@' + ver, type=('build', 'run'),
+                   when='@' + ver)
+        depends_on('rocm-device-libs@' + ver, type=('build', 'run'),
+                   when='@' + ver)
+        depends_on('llvm-amdgpu@' + ver, type=('build', 'run'),
+                   when='@' + ver)
+
+        resource(
+            name='rocm-device-libs',
+            url=compute_url +
+            '/ROCm-Device-Libs/archive/rocm-uc-' + ver + '.tar.gz',
+            sha256=versions_dict[ver]['devlib'],
+            expand=True,
+            destination='rocm-openmp-extras',
+            placement='rocm-device-libs',
+            when='@' + ver)
+
+        resource(
+            name='flang',
+            url=tools_url + '/flang/archive/rocm-uc-' + ver + '.tar.gz',
+            sha256=versions_dict[ver]['flang'],
+            expand=True,
+            destination='rocm-openmp-extras',
+            placement='flang',
+            when='@' + ver)
+
+        resource(
+            name='aomp-extras',
+            url=tools_url + '/aomp-extras/archive/rocm-uc-' + ver + '.tar.gz',
+            sha256=versions_dict[ver]['extras'],
+            expand=True,
+            destination='rocm-openmp-extras',
+            placement='aomp-extras',
+            when='@' + ver)
+
+        resource(
+            name='llvm-project',
+            url=compute_url + '/llvm-project/archive/rocm-' + ver + '.tar.gz',
+            sha256=versions_dict[ver]['llvm'],
+            expand=True,
+            destination='rocm-openmp-extras',
+            placement='llvm-project',
+            when='@' + ver)
+
+    def setup_run_environment(self, env):
+        devlibs_prefix = self.spec['rocm-device-libs'].prefix
+        openmp_extras_prefix = self.spec['rocm-openmp-extras'].prefix
+        llvm_prefix = self.spec['llvm-amdgpu'].prefix
+        env.set('AOMP', '{0}'.format(format(llvm_prefix)))
+        env.set('HIP_DEVICE_LIB_PATH',
+                '{0}/amdgcn/bitcode'.format(format(devlibs_prefix)))
+        env.set('AOMP_GPU',
+                '`{0}/rocm-bin/mygpu`'.format(format(openmp_extras_prefix)))
```

Comment:
I assume you don't want backticks in this variable

Reply:
These are intended and will set this variable to what gpu is determined and returns from mygpu. Our test suites leverage this variable.

Reply:
I'm still not sure I understand. If you run these on the command line, bash will execute the `mygpu` command. But since we aren't using bash, the entire string, including the backticks, will be passed to CMake. Is that what you want?

Reply:
The AOMP_GPU variable is intended to be used in bash when users run our examples.  I am choosing to automate this in the script to configure a user's non-build environment for convenience.  This is not intended for cmake usage.  Is using ```setup_run_environment``` not the proper way to achieve this?

Reply:
Hmm, `setup_run_environment` is the correct way to set environment variables, but to my best knowledge, this will end up being equivalent to:
```bash
export AOMP_GPU='`/path/to/rocm-bin/mygpu`'
```
Spack won't interpret the backticks, and neither will bash. If you're trying to run that command and store the output in `AOMP_GPU`, you'll want to use `Executable` instead.

Reply:
I am not familiar with ```Executable``` in Spack.

I might need to clarify a little further. The AOMP_GPU variable actually is evaluated in the Makefile of our examples as 
```
-march=$(AOMP_GPU)
``` 
which is passed on to clang. On my machine, I can verify that this returns -march=gfx906. 

If these backticks are still a problem, I could resolve with some python code instead:
```
my_gpu = subprocess.check_output(openmp_extras_prefix + '/rocm-bin/mygpu').decode(sys.stdout.encoding)
env.set('AOMP_GPU', my_gpu)
```

Reply:
Ah, I see. As long as the Makefile executes the backticks, then it's fine. Thanks for clarifying!